processor/groupbyattrsprocessor/processor.go

```diff
@@ -28,13 +28,111 @@ type groupbyattrsprocessor struct {
 
 // ProcessTraces process traces and groups traces by attribute.
 func (gap *groupbyattrsprocessor) ProcessTraces(_ context.Context, td pdata.Traces) (pdata.Traces, error) {
-	// TODO
+	rss := td.ResourceSpans()
+	for i := 0; i < rss.Len(); i++ {
+		rs := rss.At(i)
+
+		extractedResourceSpans := newSpanAttributeGroups()
+
+		ilss := rs.InstrumentationLibrarySpans()
+		for j := 0; j < ilss.Len(); j++ {
+			ils := ilss.At(j)
+			for k := 0; k < ils.Spans().Len(); k++ {
+				span := ils.Spans().At(k)
+
+				groupedAnything, groupedAttrMap, nonGroupedAttrMap := gap.splitAttrMap(span.Attributes())
+				if groupedAnything {
+					mNumGroupedSpans.M(1)
+					overwriteAttributes(nonGroupedAttrMap, span.Attributes())
+				} else {
+					mNumNonGroupedSpans.M(1)
+				}
+
+				entry := extractedResourceSpans.attributeEntry(groupedAttrMap, rs.Resource())
+				entry.instrumentationLibrarySpans(ils.InstrumentationLibrary()).Spans().Append(span)
+			}
+		}
+
+		for _, ers := range extractedResourceSpans.entries {
+			ers.resourceSpans.CopyTo(rs)
+		}
+		mDistSpanGroups.M(int64(len(extractedResourceSpans.entries)))
+	}
 
 	return td, nil
 }
 
 func (gap *groupbyattrsprocessor) ProcessLogs(_ context.Context, ld pdata.Logs) (pdata.Logs, error) {
```

Comment:
The logic of what is happening in this function is not east to understand. Perhaps add comments or re-factor for more clarity.